### PR TITLE
Attestation secureboot module

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Implement an attestation module for secureboot.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- Part of the big new doc

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Issue(s): #

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by mackdk — APPROVED** [review]

File: microservices/coco-attestation/attestation-module-secureboot/src/test/java/com/suse/coco/module/secureboot/SecureBootWorkerTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        reportMessages = Map.of(1L, "SecureBoot enabled\n",
+                2L, "SecureBoot enabled\nSecureBoot validation is disabled in shim\n",
+                3L, "SecureBoot disabled\n",
+                4L, "SecureBoot disabled\nPlatform is in Setup Mode\n",
+                5L, "Cannot determine secure boot state.\n",
+                6L, "Failed to read \"SetupMode\" variable: XYZ\n",
+                7L, "Failed to read \"SecureBoot\" variable: ABC\n"
+                );
+        // Common mocking
+        when(session.selectOne("SecureBootModule.retrieveReport", 1L)).thenReturn(reportMessages.get(1L));
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply microservices/coco-attestation/attestation-module-secureboot/src/test/java/com/suse/coco/module/secureboot/SecureBootWorkerTest.java:62]: I will try this out tomorrow

**Review by mackdk — APPROVED** [review]

**Review by deneb-alpha — APPROVED** [review]
the license header mentioning the RH copyright I suppose are needed because we are also reusing existing code. 

for the new subpackage I think we are ok. Is this also x86_64 only or are we adding this module also to aarch64 (s390x  and  ppc64le don't have shim/secureboot and I don't think it makes  any sense to have that, right)?

in any case, from what  I can read  ATM, it LGTM

**Comment by mcalmer** [comment]
@deneb-alpha the secureboot module is for all architectures. It just check for strings in the state output. This works without extra tool. Every arch can run this attestation

**Comment by deneb-alpha** [comment]
> @deneb-alpha the secureboot module is for all architectures. It just check for strings in the state output. This works without extra tool. Every arch can run this attestation

ok, thanks for the info. I thought it was working only on x86_64  and aarch64

**Pull request merged**